docs/ir/ir-breaking-changes.rst

```diff
@@ -4,3 +4,34 @@ Solidity IR-based Codegen Changes
 
 This section highlights the main differences between the old and the IR-based codegen,
 along with the reasoning behind the changes and how to update affected code.
+
+Semantic Only Changes
+=====================
+
+This section lists the changes that are semantic-only, thus potentially
+hiding new and different behavior in existing code.
+
+ * When deleting storage struct, if struct occupies less than one slot, whole slot will be cleared.
+ It is dangerous to use reminder of a slot for storing any other data as it is going to be set to zero after delete.
+
+::
+    // SPDX-License-Identifier: GPL-3.0
+    pragma solidity >0.7.0;
+
+    contract C {
+        struct S {
+            uint64 y;
+            uint64 z;
+        }
+        S s;
+        function f() public {
+            // ...
+            delete s;
+            uint256 slotValue;
```

Comment:
Maybe the part starting from her is more confusing than helpful. It could be better to describe in words that `S` only occupies 16 of the 32 bytes of the slot, but still, a `delete` will write a zero to the full slot. Please also mention that this is also done for implicit delete, e.g. if an array of structs is shortened.

Reply:
I added description in words for what happens here. Also, added that we have same behavior for implicit delete